test/ctest_helpers.cmake

```diff
@@ -204,8 +204,8 @@ endfunction()
 macro(define_c_test)
     cmake_parse_arguments(
         "C_TEST"
-        "SMOKE"
-        "TARGET;DIR_NAME;DEPENDS"
+        ""
+        "TARGET;DIR_NAME;DEPENDS;EXEC_SCRIPT"
         "SOURCES;FLAGS;ARGUMENTS"
```

Comment:
I guess line 207 is a placeholder for `one_value_keywords`, and line 208 for `multi_value_keywords`. Then what about line 209? I'm also wondering if there's a convention to use plural names for multi_value_keywords e.g. TARGETS instead of TARGET.

Reply:
Good question, just a slight correction, L207 represents options (booleans), L208 are one value keywords and L209 represents multi-value keywords e.g taking the above definition you can do
```
define_c_test(
    SMOKE <---- option TRUE/FALSE 
    TARGET myfoo <---- one value keyword, passing `myfoo`
    DIR_NAME directory_of_foo
    SOURCES "foo.c" "bar.c" <---- multi-value keyword `foo.c` & `bar.c`
)
```  
L208 shouldn't need any plurals. In some other helpers I sometimes overload the one value keywords with list values (being multiple values), though this usually means it won't be iterated or processed, rather treated as a single value e.g. dependencies for an executable definition.

Reply:
Oh great, that makes much more sense to me now. Was thinking about why TARGET is a muti_value_keyword (which we only set one value :)  It would be nice to have a comment on the blank "options" placeholder. 

Btw, does a blank "options" argument mandate the existence of the following arguments?

Reply:
Oh yep, `cmake_parse_arguments` requires 6 arguments, CMake would raise a syntax error otherwise. The option field is always expected to be the 4th argument. Hence we always need to pass an option argument. Which can be blank to indicate we aren't passing any options :)